### PR TITLE
Add ability to override jvm arguments

### PR DESCRIPTION
**Pull request by AndrewReitz**
Adds `jvm-args` to the scenario file to allow for
overriding and testing of different jvm options.

### CODE REVIEW

File: src/main/java/org/gradle/profiler/GradleScenarioInvoker.java [hunk 1]
```diff
@@ -69,7 +67,14 @@ public void doRun(GradleScenarioDefinition scenario, InvocationSettings settings
         try {
             buildConfiguration.printVersionInfo();
 
-            List<String> allBuildsJvmArgs = new ArrayList<>(buildConfiguration.getJvmArguments());
+            List<String> allBuildsJvmArgs = new ArrayList<>();
+            if (!scenario.getJvmArgs().isEmpty()) {
+                allBuildsJvmArgs.addAll(buildConfiguration.getJvmArguments().stream().filter(arg -> !arg.startsWith("-X")).collect(Collectors.toList()));
```
Comment (oehme) [inline src/main/java/org/gradle/profiler/GradleScenarioInvoker.java]: ❌ There should be no filtering logic here.

File: src/test/groovy/org/gradle/profiler/ProfilerIntegrationTest.groovy [hunk 2]
```diff
@@ -1605,4 +1605,24 @@ buildTarget {
         logFile.find("> org.gradle.profiler.number = 1").size() == 4
         logFile.find("> org.gradle.profiler.number = 2").size() == 4
     }
+
+    def "over rides jvm args"() {
```
Comment (oehme) [inline src/test/groovy/org/gradle/profiler/ProfilerIntegrationTest.groovy]: overrides

File: src/main/java/org/gradle/profiler/GradleScenarioInvoker.java (hunk 1 above)
Comment (AndrewReitz) [reply src/main/java/org/gradle/profiler/GradleScenarioInvoker.java]: Screen Shot 2019-04-11 at 7 10 54 PM

It looks like we will end up with two -Xmx (and possibly other) args if we don't filter, although the test still pass. Does the last argument always win?

**Comment by AndrewReitz** [comment]
@oehme would you like me to update the README as well?

File: src/main/java/org/gradle/profiler/GradleScenarioInvoker.java (hunk 1 above)
Comment (oehme) [reply src/main/java/org/gradle/profiler/GradleScenarioInvoker.java]: Yes that's how the jvm and most other command line programs handle arguments

**Comment by oehme** [comment]
Yes, that would be great

**Comment by AndrewReitz** [comment]
@oehme Looks like the test I added causes the build to report a failure even though all the tests pass. Do you have a suggestion for testing?

File: gradle-profiler.iml [hunk 3]
```diff
@@ -0,0 +1,13 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (oehme) [inline gradle-profiler.iml]: Can you please remove this file?

File: src/test/groovy/org/gradle/profiler/ProfilerIntegrationTest.groovy [hunk 4]
```diff
@@ -1605,4 +1605,21 @@ buildTarget {
...
         logFile.find("> org.gradle.profiler.number = 2").size() == 4
     }
+
+    def "overrides jvm args"() {
+        given:
+        def scenarios = file('performance.scenario')
+        scenarios.text = """
+help {
+    tasks = ["help"]
+    jvm-args = ["-Xmx1m"]
```
Comment (oehme) [inline src/test/groovy/org/gradle/profiler/ProfilerIntegrationTest.groovy]: The out of memory error is coming from this test - Teamcity scrapes the log, sees it and reports it as an error. I'd instead use the MemoryMxBean in the build script to print out the max memory in order to verify the setting.

Comment (oehme) [reply src/test/groovy/org/gradle/profiler/ProfilerIntegrationTest.groovy]: We could of course deactivate this check, but I think writing the test without relying on an error would be cleaner.

**Comment by oehme** [comment]
Thanks @AndrewReitz!

**Pull request merged**